dom/observable/tentative/observable-ctor.any.js

```diff
@@ -0,0 +1,685 @@
+test(() => {
+  assert_true(
+    typeof Observable === "function",
+    "Observable constructor is defined"
+  );
+}, "Observable constructor");
+
+test(() => {
+  let initializerCalled = false;
+  const results = [];
+
+  const source = new Observable((subscriber) => {
+    initializerCalled = true;
+    subscriber.next(1);
+    subscriber.next(2);
+    subscriber.next(3);
+    subscriber.complete();
+  });
+
+  assert_false(
+    initializerCalled,
+    "initializer should not be called by construction"
+  );
+
+  source.subscribe({
+    next: (x) => results.push(x),
+    error: () => assert_unreached("should not be called"),
+    complete: () => results.push("complete"),
+  });
+
+  assert_true(initializerCalled, "initializer should be called by subscribe");
+  assert_array_equals(
+    results,
+    [1, 2, 3, "complete"],
+    "should emit values synchronously"
+  );
+}, "Observable constructor calls initializer on subscribe");
+
+test(() => {
+  const error = new Error("error");
+  const results = [];
+
+  const source = new Observable((subscriber) => {
+    subscriber.next(1);
+    subscriber.next(2);
+    subscriber.error(error);
+  });
+
+  source.subscribe({
+    next: () => assert_unreached("should not be called"),
+    error: (e) => results.push(e),
+    complete: () => assert_unreached("should not be called"),
+  });
+
+  assert_array_equals(
+    results,
+    [1, 2, error],
+    "should emit error synchronously"
+  );
+}, "Observable error path called synchronously");
+
+test(() => {
+  const error = new Error("error");
+  const results = [];
+
+  const source = new Observable((subscriber) => {
+    try {
+      subscriber.next(1);
+      throw error;
+    } finally {
+      assert_true(subscriber.closed, "subscriber is closed after error");
+    }
+  });
+
+  source.subscribe({
+    next: (x) => results.push(x),
+    error: (e) => results.push(e),
+    complete: () => assert_unreached("should not be called"),
+  });
+
+  assert_array_equals(
+    results,
+    [1, error],
+    "should emit values and the throw error synchronously"
+  );
+}, "Observable should error if initializer throws");
+
+test(() => {
+  let initializerCalled = false;
+  const source = new Observable(() => {
+    initializerCalled = true;
+  });
+
+  assert_false(
+    initializerCalled,
+    "initializer should not be called by construction"
+  );
+  source.subscribe();
+  assert_true(initializerCalled, "initializer should be called by subscribe");
+}, "subscribe can be called with no arguments");
+
+test(() => {
```

Comment:
Actually this (and the corresponding error test) is actually a tiny bit broken I think. If `subscriber.complete()` fails (because it is not implemented), then no error will be thrown and no assertion will ever be run. Can we instead assert that `subscriber.complete` exists as a function, so incomplete implementations will fail more completely and more specifically? This comment probably applies to other tests in this file as well, where we can assert that many more functions/members exist, since the baseline semantics don't throw an error, so an incomplete test may incorrectly pass a bunch of tests.

Reply:
OK I'm less worried about this now that I added one general subscriber-shape test in https://github.com/web-platform-tests/wpt/pull/42219/files#diff-9ca50b395b5363c92235bb59bfe0859d1258001d3828f62b93fccfe4ed28d887R12-R27, so maybe we can just resolve this.